### PR TITLE
cbc split code

### PR DESCRIPTION
**Pull request by thouis**

### CODE REVIEW

File: donor_assignment/likelihood_per_region.py [hunk 1]
```diff
@@ -94,55 +94,60 @@ def main():
...
-    #
-    # Add in umi counts and snp counts
-    #
-
-    num_snps = umi_probs_position_index.groupby(['barcode']).size()
-    num_umis = umi_probs_position_index.reset_index().groupby('barcode')['UMI'].unique().str.len()
-    barcode_log_likelihood['num_snps'] = num_snps
-    barcode_log_likelihood['num_umis'] = num_umis
-
+    unique_barcodes = np.unique(single_base_uniq_reads.reset_index('barcode').barcode)
```
Comment (thouis) [inline donor_assignment/likelihood_per_region.py]: Why np.unique() vs. (...).barcode.unique()

File: donor_assignment/likelihood_per_region.py [hunk 2]
```diff
@@ -94,55 +94,60 @@ def main():
...
+    barcode_lls = []
+    barcode_reads = single_base_uniq_reads.copy().reset_index('barcode')
```
Comment (thouis) [inline donor_assignment/likelihood_per_region.py]: Is this copy() necessary?

File: donor_assignment/likelihood_per_region.py [hunk 3]
```diff
@@ -94,55 +94,60 @@ def main():
...
+    all_cbcs = unique_barcodes.copy()
```
Comment (thouis) [inline donor_assignment/likelihood_per_region.py]: combine with unique_barcodes

File: donor_assignment/likelihood_per_region.py [hunk 4]
```diff
@@ -94,55 +94,60 @@ def main():
...
+    while len(all_cbcs) > 0:
+        cur_cbcs = all_cbcs[:100]
+        all_cbcs = all_cbcs[100:]
+        barcode_pos_reads = barcode_reads[barcode_reads.barcode.isin(cur_cbcs)].reset_index('pos')
```
Comment (thouis) [inline donor_assignment/likelihood_per_region.py]: would be clearer to pull code below this into a separate function, possibly.

File: donor_assignment/likelihood_per_region.py (hunk 3 above)
Comment (thouis) [reply donor_assignment/likelihood_per_region.py]: Also sort(), and wrap `while` with a `with gzip.open(...) as outf:` and remove barcode_lls above

File: donor_assignment/likelihood_per_region.py [hunk 5]
```diff
@@ -94,55 +94,60 @@ def main():
...
+        barcode_log_likelihood = regularized_log_probs.groupby(['barcode'])[donors].sum()
+        num_snps = umi_probs_position_index_100cbcs.groupby(['barcode']).size()
+        num_umis = umi_probs_position_index_100cbcs.reset_index().groupby('barcode')['UMI'].unique().str.len()
+        barcode_log_likelihood['num_snps'] = num_snps
+        barcode_log_likelihood['num_umis'] = num_umis
+        
+        # append current barcode lls to total list
+        barcode_lls.append(barcode_log_likelihood)
+
+    final_barcode_log_likelihoods = pd.concat(barcode_lls)
```
Comment (thouis) [inline donor_assignment/likelihood_per_region.py]: Wrapping with a `with gzip.open()` will remove this concat, free up memory

File: donor_assignment/likelihood_per_region.py (hunk 1 above)
Comment (murphymadeleine21) [reply donor_assignment/likelihood_per_region.py]: For some reason didn't look like barcode.unique() was actually giving the unique values

File: donor_assignment/likelihood_per_region.py (hunk 2 above)
Comment (murphymadeleine21) [reply donor_assignment/likelihood_per_region.py]: We did this before, and decided it was best

File: donor_assignment/likelihood_per_region.py (hunk 1 above)
Comment (thouis) [reply donor_assignment/likelihood_per_region.py]: It should be the same.  Maybe test to be sure, but if .unique() isn't doing what we expect then there are deeper problems.

File: donor_assignment/likelihood_per_region.py (hunk 2 above)
Comment (thouis) [reply donor_assignment/likelihood_per_region.py]: I think we can take it out, now, since we're not adding columns to the DF anymore

**Pull request closed without merging**